Update AnalogEncoder and DutyCycleEncoder for 2025 (#2943)
wpilibsuite/frc-docs: source/docs/software/hardware-apis/sensors/encoders-software.rst

The roboRIO's FPGA handles duty cycle encoders automatically.

> .. warning:: The API has changed for 2025 to remove rollover detection as rollover detection did not work. The :code:`get()` method returns the value within a rotation where the maximum value in a rotation is defined in the constructor (default 1).

Reviewer: Can we write this "future proof"
```suggestion
.. warning:: In 2025 the API changed to remove rollover detection as rollover detection did not work. The :code:`get()` method returns the value within a rotation where the maximum value in a rotation is defined in the constructor (default 1).
```

    frc::DutyCycleEncoder encoder{0};
    ```

### Configuring Duty Cycle Encoder Range and Zero

> .. note:: The :code:`DutyCycleEncoder` class does not make any assumptions about units of distance; it will return values in whatever units were used to calculate the full range value. Users thus have complete control over the distance units used.

Reviewer: ```suggestion
.. note:: The :code:`DutyCycleEncoder` class does not assume specific units of distance. It returns values in the same units used to calculate the full range, giving users complete control over the distance units.
```

> The :code:`DutyCycleEncoder` class offers an alternate constructor that offers control over the distance per rotation and zero position of the encoder.

Reviewer: Double offers sounds strange to me.
```suggestion
The :code:`DutyCycleEncoder` class provides an alternate constructor that allows control over the distance per rotation and the zero position of the encoder.
```

> The zero is useful for ensuring that the measured rotation corresponds to the actual desired physical measurement. Unlike quadrature encoders, duty cycle encoders don't need to be homed. The current rotation that is desired to be read and then stored to be set when the program starts. The :doc:`Preferences class </docs/software/basic-programming/robot-preferences>` provides a method to save and retrieve the values on the roboRIO.

Reviewer: ```suggestion
The zero position is useful for ensuring that the measured rotation corresponds to the desired physical measurement. Unlike quadrature encoders, duty cycle encoders don't need to be homed. The desired rotation can be read and stored to be set when the program starts. The :doc:`Preferences class </docs/software/basic-programming/robot-preferences>` provides methods to save and retrieve these values on the roboRIO.
```

.. tab-set-code::

    ```java
    // Initializes an analog encoder on Analog Input pins 0
    AnalogEncoder encoder = new AnalogEncoder(0);
    ```

    ```c++
    // Initializes a analog cycle encoder on DIO pins 0
    frc::AnalogEncoder encoder{0};
    ```

### Configuring Analog Encoder Range and Zero

> .. note:: The :code:`AnalogEncoder` class does not make any assumptions about units of distance; it will return values in whatever units were used to calculate the full range value. Users thus have complete control over the distance units used.

Reviewer: ```suggestion
.. note:: The :code:`AnalogEncoder` class makes no assumptions about distance units, returning values in the same units used to calculate the full range. This gives users complete control over the choice of distance units.
```

The :code:`AnalogEncoder` class offers an alternate constructor that offers control over the distance per rotation and zero position of the encoder.

> The zero is useful for ensuring that the measured rotation corresponds to the actual desired physical measurement. Unlike quadrature encoders, analog encoders don't need to be homed. The current rotation that is desired to be read and then stored to be set when the program starts. The :doc:`Preferences class </docs/software/basic-programming/robot-preferences>` provides a method to save and retrieve the values on the roboRIO.

Reviewer: ```suggestion
The zero position is useful for ensuring that the measured rotation corresponds to the desired physical measurement. Unlike quadrature encoders, analog encoders don't need to be homed. The desired rotation can be read and stored to be set when the program starts. The :doc:`Preferences class </docs/software/basic-programming/robot-preferences>` provides methods to save and retrieve these values on the roboRIO.
```

.. tab-set-code::

    ```java
    // Initializes an analog encoder on DIO pins 0 to return a distance of 4 for
    // every rotation, with the encoder reporting 0 half way through rotation (2
    // out of 4)
    AnalogEncoder encoder = new AnalogEncoder(0, 4.0, 2.0);
    ```

    ```c++
    // Initializes an analog encoder on DIO pins 0 to return a distance of 4 for
    // every rotation, with the encoder reporting 0 half way through rotation (2
    // out of 4)
    frc::AnalogEncoder encoder{0, 4.0, 2.0};
    ```

### Reading Distance from Analog Encoders

> .. note:: Analog encoders measure absolute distance. It does not depend on the starting position of the encoder.

Reviewer: This might be confusing since we are changing everything else to rotation
